lib/link_state.c

```diff
@@ -417,6 +404,25 @@ int ls_prefix_same(struct ls_prefix *p1, struct ls_prefix *p2)
 /**
  *  Link State Vertices management functions
  */
+uint64_t sysid_to_key(const uint8_t sysid[ISO_SYS_ID_LEN])
```

Comment:
Should we not store ISO system-Ids in Network-Byte-Order irrespective of endian-ness? Please excuse my ignorance if I did not get it right. In my experience outside of FRR, everywhere the IPv4/v6 addresses and ISO mysid/lspid/NET were always stored locally as network-byte-order.

Reply:
That's a great question. Of course it should. But, the ISO sys ID as well as IP addresses are used to compute the key of the vertex and edge in the graph. If you let them in network byte order, this ended to a non contiguous, large number, non human readable key when you print the TED on a little endian machine while it is OK on a big endian arch. Thus, I prefer to convert the key into endian order of the machine to get an homogeneous representation of TED. This is also necessary for test as printing TED in json format on an little endian and big endian machine will not give the same result instead of IP address which are converted before printing.